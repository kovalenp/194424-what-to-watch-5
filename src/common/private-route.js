import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {authStatus, appRoute} from "./constants";


const PrivateRoute = (props) => {
  const {render, path, exact, isAuth} = props;

  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        return (
          isAuth
            ? render(routeProps)
            : <Redirect to={appRoute.LOGIN} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  isAuth: PropTypes.bool.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  return {
    isAuth: state.user.authentication === authStatus.AUTH
  };
};

export default connect(mapStateToProps)(PrivateRoute);
