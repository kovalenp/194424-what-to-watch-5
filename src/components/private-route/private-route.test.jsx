import React from "react";
import renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route";
import {MemoryRouter} from 'react-router-dom';

it(`PrivateRoute test to meet B21 criteria`, () => {
  const tree = renderer
    .create(
        <MemoryRouter>
          <PrivateRoute />
        </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
