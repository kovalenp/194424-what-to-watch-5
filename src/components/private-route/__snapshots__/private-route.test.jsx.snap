// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute test to meet B21 criteria 1`] = `null`;
